Fix message of failed sync

jsafrane opened the pull request:
Message `couldn't propagate object cache` ends up in Event about failed mount and it looks confusing to users. Trying `failed to sync X cache` (where X is either `secret` or `configmap`).

From:

`MountVolume.SetUp failed for volume "secret-prometheus-k8s-proxy" : couldn't propagate object cache: timed out waiting for the condition`

    
To:
`MountVolume.SetUp failed for volume "secret-prometheus-k8s-proxy" : failed to sync secret cache: timed out waiting for the condition
`

/kind cleanup
/sig node

```release-note
NONE
```

feiskyer reviewed (commented):
/lgtm

poothia commented:
lgtm

jsafrane commented:
/assign @tallclair @derekwaynecarr 
for approval

xichengliudui commented:
I may have the same problem as you: `couldn't propagate object cache: timed out waiting for the condition`.
there is an exception log in kubelet. ref: https://github.com/xichengliudui/public-repo/blob/master/logs.md

xichengliudui commented:
Are they the same?

tallclair commented on pkg/kubelet/util/manager/watch_based_manager.go:
```diff
@@ -166,7 +166,7 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 	}
 	if err := wait.PollImmediate(10*time.Millisecond, time.Second, item.hasSynced); err != nil {
-		return nil, fmt.Errorf("couldn't propagate object cache: %v", err)
+		return nil, fmt.Errorf("failed to sync %s cache: %v", c.groupResource.Resource, err)
```
```suggestion
		return nil, fmt.Errorf("failed to sync %s cache: %v", c.groupResource.String(), err)
```

jsafrane replied:
fixed

jsafrane committed:
Fix message of failed sync

From:
MountVolume.SetUp failed for volume "secret-prometheus-k8s-proxy" :
couldn't propagate object cache: timed out waiting for the condition

To:
MountVolume.SetUp failed for volume "secret-prometheus-k8s-proxy" : failed
to sync secret cache: timed out waiting for the condition

poothia commented:
i think this issue can be closed.

jsafrane commented:
I forgot about this one .I still think this PR is useful, @tallclair, would you please check again?

tallclair commented:
/lgtm
/approve
/retest
